aspnetcore/fundamentals/routing.md

```diff
@@ -26,41 +26,42 @@ Apps can configure routing using:
 * For Razor Pages conventions, see <xref:razor-pages/razor-pages-conventions>.
 
-The endpoint routing system described in this document applies to ASP.NET Core 3.0 and later. For information on the previous routing system based on <xref:Microsoft.AspNetCore.Routing.IRouter>, select the ASP.NET Core 2.1 version using one of the following approaches:
-
-* The version selector for a previous version.
-* Select [ASP.NET Core 2.1 routing](?preserve-view=true&view=aspnetcore-2.1).
-
-[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/fundamentals/routing/samples/3.x) ([how to download](xref:index#how-to-download-a-sample))
-
-The download samples for this document are enabled by a specific `Startup` class. To run a specific sample, modify *Program.cs* to call the desired `Startup` class.
-
 ## Routing basics
 
-All ASP.NET Core templates include routing in the generated code. Routing is registered in the [middleware](xref:fundamentals/middleware/index) pipeline in `Startup.Configure`.
-
 The following code shows a basic example of routing:
 
-:::code language="csharp" source="routing/samples/3.x/RoutingSample/Startup.cs" id="snippet" highlight="8,10":::
+:::code language="csharp" source="routing/samples/6.0/RoutingSample/Program.cs" highlight="4":::
+
+The preceding example includes a single endpoint using the [MapGet](xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapGet%2A) method:
+
+* When an HTTP `GET` request is sent to the root URL `/`:
+  * The request delegate executes.
+  * `Hello World!` is written to the HTTP response.
+* If the request method is not `GET` or the root URL is not `/`, no route matches and an HTTP 404 is returned.
 
 Routing uses a pair of middleware, registered by <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseRouting%2A> and <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseEndpoints%2A>:
 
 * `UseRouting` adds route matching to the middleware pipeline. This middleware looks at the set of endpoints defined in the app, and selects the [best match](#urlm) based on the request.
 * `UseEndpoints` adds endpoint execution to the middleware pipeline. It runs the delegate associated with the selected endpoint.
 
-The preceding example includes a single *route to code* endpoint using the [MapGet](xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapGet%2A) method:
+Apps that use [Minimal APIs](xref:fundamentals/minimal-apis) typically don't need to call `UseRouting` or `UseEndpoints`. <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder> configures a middleware pipeline that wraps middleware added in *Program.cs* with `UseRouting` and `UseEndpoints`. However, apps that use Minimal APIs can change the order in which `UseRouting` and `UseEndpoints` run by calling these methods explicitly. For example, fhe following code makes an explicit call to `UseRouting`:
```

Comment:
Shouldn't this change this to ASP.NET Core apps? That applies to all apps. I don't want to give the impression you need to call UseRouting or UseEndpoints.  in other apps.

Reply:
I dunno... We're in the context of ASP.NET Core here, so I thought it might be redundant to state that here. I can add it if you think it helps, of course. Just make the suggestion and I'll commit it with the other one.

Reply:
I think someone could infer minimal api's were different given that's what you're calling out. How about

```suggestion
ASP.NET Core apps typically don't need to call `UseRouting` or `UseEndpoints`. <xref:Microsoft.AspNetCore.Builder.WebApplicationBuilder> configures a middleware pipeline that wraps middleware added in *Program.cs* with `UseRouting` and `UseEndpoints`. However, apps can change the order in which `UseRouting` and `UseEndpoints` run by calling these methods explicitly. For example, fhe following code makes an explicit call to `UseRouting`:
```

Reply:
Ok, I think I understand what you're going for now. Can we just say "Apps typically don't need..."? That doesn't call out Minimal APIs as being special, and also doesn't repeat ASP.NET Core that's already a given imo.